src/main/java/org/elasticsearch/gateway/GatewayAllocator.java

```diff
@@ -118,8 +120,8 @@ public boolean allocateUnassigned(RoutingAllocation allocation) {
             if (!routingNodes.routingTable().index(shard.index()).shard(shard.id()).primaryAllocatedPostApi()) {
                 continue;
             }
-            final String indexUUID = allocation.metaData().index(shard.index()).getUUID();
-            ObjectLongOpenHashMap<DiscoveryNode> nodesState = buildShardStates(nodes, shard, indexUUID);
+
+            ObjectLongOpenHashMap<DiscoveryNode> nodesState = buildShardStates(nodes, shard, metaData.index(shard.index()));
```

Comment:
actually I think it's ok - I initially passed more stuff on here to resolve custom paths but the state is on the node private paths so I reverted the actual change...